### PR TITLE
Show direct match on top for user search

### PR DESCRIPTION
**Pull request by maweil**
This PR makes sure that direct matches in the user search always show on top of the result list.
This fixes issue #14350 which was originally reported on codeberg.

The following places were checked to follow the desired behavior now:
- Search when adding a user to a team
- Search when adding a user as a collaborator to a repository

The following checks were performed successfully:
- `make lint-frontend`
- `make test`

Any feedback is welcome. Thanks.

### CODE REVIEW

**Review by wxiaoguang — APPROVED** [review]

File: web_src/js/index.js [hunk 1]
```diff
@@ -2233,15 +2233,21 @@ function searchUsers() {
...
           if (item.full_name && item.full_name.length > 0) {
             title += ` (${htmlEscape(item.full_name)})`;
           }
-          items.push({
+          const resultItem = {
             title,
             image: item.avatar_url
-          });
+          };
+          if (searchQuery === item.login) {
```
Comment (lunny) [inline web_src/js/index.js]: or it should be case sensitive.

Comment (maweil) [reply web_src/js/index.js]: Not sure if I understood correctly. Do you mean it should compare ignoring the case differences?
That would make sense. I'll have a look.

Comment (maweil) [reply web_src/js/index.js]: I updated the PR to ignore case differences as shown here:
Screenshot search ignoring case

**Review by noerw — APPROVED** [review]

**Comment by 6543** [comment]
:rocket:

**Pull request merged**